cupy/random/distributions.py

```diff
@@ -438,9 +437,8 @@ def negative_binomial(n, p, size=None, dtype=int):
     return rs.negative_binomial(n, p, size=size, dtype=dtype)
 
 
-def multivariate_normal(mean, cov, size=None, check_valid='ignore', tol=1e-8,
```

Comment:
the `check_valid` and `tol` args are needed to keep compatibility with numpy

https://docs.scipy.org/doc/numpy/reference/random/generated/numpy.random.Generator.multivariate_normal.html?highlight=multivariate_normal#numpy.random.Generator.multivariate_normal

Reply:
this should be fixed now. Since the `check_valid` option is in place, I removed the other warning... in numpy, cholesky will fail non-silently on a non-semipositive definite matrix, so `check_valid` doesn't have much meaning there. I'm not sure of the behavior for cupy, but the warning attached to cholesky about conditions needing to be met makes me think that the routine doesn't explicitly check the state of the matrix with regards to semipositive definiteness (hence the option of `cupyx.errstate()` or `cupyx.seterr()` to explicitly warn on that decomposition). The way things are set now, is that there should be a warning regardless of the state of `check_valid` if `cupyx.errstate()` or `cupyx.seterr()` is invoked; if they are not, and `check_valid` is set to anything other than 'ignore', an extra eigh call will test that the covariance matrix input supplied is valid. Cholesky should always be faster than svd, and eigh is slightly slower than cholesky...but since this wasn't working earlier, even if `check_valid` is used, the routine will now at least run and complete.

Reply:
Yes, I agree with what you say and totally understand it. 😄
However, using cholesky inside this function is an implementation detail and the interface exposed to the users should not deviate from the numpy one.
For example cupy fails silently with cholesky
```
>>> a=cupy.arange(9).reshape(3,3)
>>> cupy.linalg.cholesky(a)
array([[nan,  0.,  0.],
       [inf, nan,  0.],
       [inf, nan, nan]])
```
We should try to deal with cholesky error states inside this function and then adapt them to what the interface requires, even if its redundant as we strive towards compatibility.
Sorry to keep asking for changes 😔